crates/sparrow-session/src/execution.rs

```diff
@@ -56,13 +54,13 @@ impl Execution {
     /// status (and return) accordingly.
     fn is_done(&mut self) -> error_stack::Result<(), Error> {
         let result = match &mut self.status {
-            Status::Running(future) => {
+            Status::Running(progress) => {
                 // Based on the implementation of `FutureExt::now_or_never`:
                 let noop_waker = futures::task::noop_waker();
                 let mut cx = std::task::Context::from_waker(&noop_waker);
 
-                match future.as_mut().poll(&mut cx) {
-                    std::task::Poll::Ready(x) => x,
+                match progress.try_poll_next_unpin(&mut cx) {
```

Comment:
I'm pretty sure this is the equivalent behavior

Reply:
I'm not sure. I don't think so -- previously it ran in a loop to wait until the end of the stream. Now it seems to return sooner.